tasks/main.yml

```diff
@@ -25,32 +25,59 @@
     enabled: yes
   become: yes
 
-- name: Install Infiniband packages
+- name: Install Gluster packages
   package:
     name: "{{ item }}"
     state: present
+  with_items:
+    - glusterfs-server
+    - glusterfs-client
+    - glusterfs-rdma
+    - glusterfs-fuse
   become: yes
+
+- name: Install additional packages
+  package:
+    name: "{{ item }}"
+    state: present
   with_items:
     - '@InfiniBand Support'
     - perftest
     - infiniband-diags
+    - mdadm
+  become: yes
 
-- name: Install Gluster packages
-  package:
-    name: "{{ item }}"
-    state: present
-    - glusterfs-server
-    - glusterfs-client
-    - glusterfs-rdma
-    - glusterfs-fuse
+- name: Set gluster_cluster_devices to gluster_cluster_block_devices
```

Comment:
I'm a bit confused about gluster_cluster_devices, whether RAIDed or not it's still a list of block devices?
I think I would also rather see this as a separate role, if we need one. I imagine the ansible logical volume manager module will get you pretty far. Once that role has run, just pass the RAIDed block devices to this role.